pep-9999.rst

```diff
@@ -0,0 +1,520 @@
+  release would be considered provisional
+* as with current alpha releases, beta releases would be prepared and published
+  from the master development branch
+
+
+Rather than being released monthly for a period of a few months while preparing
+a new X.Y.0 release, beta releases would instead be consistently published every
+other month.
+
+The only time this would not be the case is during the two month release
+candidate period for an upcoming X.Y.0 release.
+
+
+Changes to release candidate policy, phase duration, and cadence
+----------------------------------------------------------------
+
+Given the proposed changes to the beta release phase, the release candidate
+phase would see the following related adjustments:
+
+* Feature freeze, ABI freeze, pyc file format freeze, and maintenance branch
+  creation would all correspond with the creation of X.Y.0rc1 (currently these
+  occur across a mixture of X.Y.0b1, the last beta release, and X.Y.0rc1)
+* The X.Y.0 release candidate period would be extended from 3 weeks to 2 months
+* There would normally be two release candidates issued a month apart, but
+  additional candidates may be published at the release manager's discretion
+* The final release would occur between 1 and 4 weeks after the final release
+  candidate (depending if additional release candidates were needed after the
+  second)
+
+
+Removal of routine alpha releases
+---------------------------------
+
+Due to the proposed policy changes for beta releases and release candidates,
+CPython alpha releases would no longer serve a clear purpose, and as such
+would typically be skipped.
+
+If they chose to, release managers would still be free to make one or more
+alpha releases before entering the rolling beta release period. For example,
+that would be one way of testing out new release management tooling.
+
+
+Changes to extension module loading in CPython pre-releases
+-----------------------------------------------------------
+
+Extension modules built against CPython pre-releases prior to the ABI freeze
+for that release series may not actually load correctly on the final release.
+
+Historically, adoption of alpha and beta releases has been low enough that this
+hasn't really been a problem in practice. However, this PEP proposes to actively
+encourage widespread operational use of beta releases, and we'd like to make it
+clear that such binaries are only reliably portable to the exact version that
+was used to build them (they may fail to load on even the very next alpha or
+beta release).
+
+To that end, this PEP proposes amending the extension module ``SOABI`` marker
+for CPython pre-releases to use a "cpython_unstable-" prefix, and only switch
+back to the regular "cpython-" prefix once the ABI for that particular version
+has been frozen on entry to the release candidate stage.
+
+(TODO: Link to https://github.com/ncoghlan/cpython/pull/3)
+(TODO: What's the Windows equivalent of this?)
```

Comment:
The Windows equivalent would be here:

https://github.com/python/cpython/blob/9c42f8cda552694f3b47d6388d4ae84d61731872/PCbuild/python.props#L208-L211

The debug flag goes into the main part of the filename as a `_d` suffix, which is why it's not here, and we've never had to worry about separating by other ABI flags.